elkscmd/ash/redir.c

```diff
@@ -232,7 +231,7 @@ openhere(redir)
 		error("Pipe call failed");
 	if (redir->type == NHERE) {
 		len = strlen(redir->nhere.doc->narg.text);
-		if (len <= PIPESIZE) {
+		if (len <= PIPE_BUF) {
```

Comment:
Nice catch on ELKS' max pipe buffering size. I hadn't realized the shell would fork in the case a 'here document' greater than the max writable pipe buffer. However, the kernel will still accept pipe writes greater than the pipe buffer size by sleeping the process and waiting for the reader to deplete the buffer. I haven't checked enough to see whether this might deadlock the shell or not, assuming that ash is likely also the reader (?). Thus, we should probably test this change with a small and large here document, perhaps in the test suite. 

Thus, this is either pointing out a previously unknown bug in the shell for ELKS, or likely running new untested fork code for the shell. The problem with the shell forking is that its large and there may not be space for the extra data segment (text segment is shared).

Reply:
I have previously tested very large here documents, and just re-tested ~300 bytes, and I am able to see that it forks.  I have never been able to cause a deadlock.